src/components/tx/SignOrExecuteForm/index.tsx

```diff
@@ -248,9 +235,7 @@ const SignOrExecuteForm = ({
         />
 
         {/* Error messages */}
-        {isWrongChain ? (
-          <ErrorMessage>Please connect your wallet to {currentChain?.chainName}</ErrorMessage>
-        ) : cannotPropose ? (
+        {isSubmittable && cannotPropose ? (
```

Comment:
This prevents the error shown between hardware wallet network switching explained [here](https://github.com/safe-global/web-core/pull/1746#issuecomment-1470284593).